src/main/java/com/example/onlinebookstore/repository/AbstractRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.example.onlinebookstore.repository;
+
+import jakarta.persistence.EntityManagerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public abstract class AbstractRepository {
```

Comment:
this abstraction doesn't bring any benefits - it just adds up complexity

extracting a single constructor to and abstract class is not the best idea :)